Add delete

peterdgreat opened the pull request:
## A simple CRUD functionality added 
- An Input text added
- Input value is updated both in dom, array, and local storage on clicking enter 
- Remove function created which remove selected task elements
- Clear all tasks on the topmost right which delete all elements
- Clear all completed which deletes all completed tasks
- a background color on editing added

peterdgreat committed:
Input text value added

- input value from tasks.js
-loacal storage class moved to tasks.js
-id added using length of array

peterdgreat committed:
add- delete

- p content editable on click
- icon changed to bin on clicked
-changed items updated on local storage
- delete function added
-icon deleted both from dom and local storage

peterdgreat committed:
Clear All Completed

- All completed are cleared in a single click

peterdgreat committed:
Todo styled

- background color added on edit

peterdgreat committed:
Linter fix

MahmoudBakr23 commented on src/tasks.js:
```diff
@@ -1,19 +1,167 @@
+          more.innerText = 'delete';
+          more.addEventListener('click', () => {
+            // delete selected item from dom
+            li.remove();
```
- [ ] Please double-check on the removing functionality for a single task because it seems to not responding to that event correctly. 😄 🙏 I can only remove the completed task using the button!

peterdgreat committed:
Functionality update

- Refresh updates and users can add after reereshes
-(required change)Delete button tested, updated and improved

MahmoudBakr23 commented on src/tasks.js:
```diff
@@ -1,19 +1,167 @@
+    }
+
+    static deleteCompleted() {
+      const domStay = document.querySelectorAll('.completed');
```
- [ ] **OPTIONAL**
The 'delete all completed' button is within the `<ul>` tag and this is semantically not right, don't you think it'd be better if we moved the button out of the tasks list because it's not a task right? 😄 Please tell me what you think!

peterdgreat replied:
It looks nice, but this is not me selecting the whole ul, this is me selecting the whole list that are completed

peterdgreat replied:
However I implemented your suggestions, and it looks better

MahmoudBakr23 commented on src/tasks.js:
```diff
@@ -1,19 +1,167 @@
+    }
+}
+
+refresh.addEventListener('click', () => {
+  Tasks.deleteAllTasks();
+  ul.remove();
+});
```
- [ ] **OPTIONAL**
Extra credit for implementing a use for the refresh button. However, it seems that you're removing the whole list element from the app when this event is fired, and this is preventing users from adding tasks after! It'd be great if you worked on that either 😄 😉

peterdgreat replied:
Hello I already resolved all the required changes you highlighted, I only left a few of the optional to do at my free time, you can double check

MahmoudBakr23 reviewed (changes requested):
Hello @peterdgreat 😃,

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linter is correct
- Descriptive PR summary and commit messages
- Clean code

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

- [ ] Please write a more descriptive PR headline to make it easier for viewers and reviewers to quickly catch up on what this PR is about! 😄 🙏 _Kindly keep in mind that this issue would normally get your PR invalidated._

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] Kindly consider attaching an image of your app to the README file. This would make the repo more professional and give it better looking. 😄 
- [ ] Also, please update your live demo link to the app's latest updates for consistency and professionalism matters. 😄 

_Please note that the README file isn't loaded for inline comments._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

MahmoudBakr23 reviewed (changes requested):
Hi @peterdgreat 😃,

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

- [ ] Please make the changes from the previous review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

peterdgreat committed:
Remove functionality on a single task updated

- I update the index of the items when tasks are added and deleted from the arrays
- this helps to monitor changes and update on DOM
-  I implemented delete-all selected outside the Ul, as it looks so better as suggested by the last reviewer

peterdgreat committed:
README Updated

--added a screenshot of the web

Emmyn5600 reviewed (approved):
Hi,

Good job so far! 👏🏼 @peterdgreat 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- The project looks awesome:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous reviewer.

 #### Optional suggestions
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
